### PR TITLE
Preserve command line ini settings for restart

### PR DESCRIPTION
**Pull request by johnstevenson**
This is a more robust fix for #6004 as it handles any settings from the command line.

### CODE REVIEW

File: src/Composer/XdebugHandler.php [hunk 1]
```diff
@@ -165,19 +166,56 @@ private function writeTmpIni(array $iniFiles)
...
+            if ($value === null || strpos($name, 'xdebug') === 0) {
+                continue;
+            }
+
+            if (!isset($iniConfig[$name]) || $iniConfig[$name] !== $value) {
+                // Based on main -d option handling in php-src/sapi/cli/php_cli.c
+                if ($value && !ctype_alnum($value)) {
+                    $value = '"'.str_replace('"', '\\"', $value).'"';
+                }
+                $content .= $name.'='.$value.PHP_EOL;
```
Comment (Seldaek) [inline src/Composer/XdebugHandler.php]: HHVM complains about an array to string conversion here. I guess some ini values can be arrays (maybe only in hhvm?) Anyway might be worth guarding against this..

**Comment by Seldaek** [comment]
I do appreciate the work, but this PR adds 3-400ms of init time to composer for me.. if I compare before/after with `time composer -h`. I wonder if it is really worth it given it really just fixes the edge case of someone passing $randomFlagWeCantThinkOf to php.

**Comment by Seldaek** [comment]
Given the impact I think I'd rather hold off on a merge until we see that this is needed (i.e. if people come up with new ini settings they use and we need to add every other week).

**Comment by johnstevenson** [comment]
I'm surprised it adds that much time. I did worry about this but testing in a vm Ubuntu setup didn't seem to make much difference. On Windows it takes forever to create a new process anyway.

No worries about dropping this. The hhvm array thing is interesting because PHP's `ini_get_all()` specifically ignores array ini options. I'll fix this bit in case we should ever want to use it.

**Comment by Seldaek** [comment]
Think we can close this for now :)

**Pull request closed without merging**